packages/Chem/src/widgets/scaffold-tree.ts

```diff
@@ -1512,3 +1544,56 @@ class SketcherDialogWrapper {
   static SUCCESS_MSG_COLOR = 'green';
   static FAILURE_MSG_COLOR = 'red';
 }
+
+export class ScaffoldTreeFilter extends DG.Filter {
+  viewer: ScaffoldTreeViewer = new ScaffoldTreeViewer();
+  savedTree: string = '';
+  
+  constructor() {
+    super();
+    this.root = ui.divV([]);
+    this.subs = [];
+  };
+
+  get isFiltering(): boolean {
+    return super.isFiltering;
+  };
+
+  get filterSummary(): string {
+    return ScaffoldTreeViewer.TYPE;  
+  };
+
+  attach(dataFrame: DG.DataFrame): void {
+    super.attach(dataFrame);
+    this.column ??= dataFrame.columns.bySemType(DG.SEMTYPE.MOLECULE);
+    this.columnName ??= this.column?.name;
+    this.createViewer(dataFrame);
+  };
+
+  saveState(): any {
+    const state = super.saveState();
+    state.savedTree = JSON.stringify(ScaffoldTreeViewer.serializeTrees(this.viewer.tree));
+    this.viewer.updateFilters(this.isFiltering);
```

Comment:
Why would you update filters when saving the state?

Reply:
Fixed